pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -73,6 +81,26 @@ func (cm RevisionRequestConcurrencyModelType) Validate() *apis.FieldError {
 	}
 }
 
+func ValidateContainerConcurrency(cc RevisionContainerConcurrencyType, cm RevisionRequestConcurrencyModelType) *apis.FieldError {
+	// Validate ContainerConcurrency alone
+	if cc < 0 || cc > RevisionContainerConcurrencyMax {
+		return apis.ErrInvalidValue(strconv.Itoa(int(cc)), "containerConcurrency")
+	}
+
+	// Validate combinations of ConcurrencyModel and ContainerConcurrency
```

Comment:
I think you can simplify this to:

```golang
if cm == RevisionRequestConcurrencyModelType("") {
  return nil
}
...
```

Won't it be possible for `cc` to be 0 and `cm` to be single if you receive a legacy revision which only uses the `concurrencyModel` field?